### PR TITLE
Stream State Update

### PR DESCRIPTION
**Pull request by hiroshihorie**
Flutter ver of https://github.com/livekit/client-sdk-ios/pull/29

### CODE REVIEW

**Review by boks1971 — APPROVED** [review]
Looks good @hiroshihorie .

I am unable to understand if this firing event at room level :-) Is it?

@davidliu When you get a chance, can you confirm if you are firing these at room level in the Android case?

**Comment by hiroshihorie** [comment]
Thanks oniichann!! 😁🙏

File: lib/src/publication/remote_track_publication.dart [hunk 1]
```diff
@@ -40,6 +41,27 @@ class RemoteTrackPublication<T extends RemoteTrack>
...
+  /// paused / resumed by the server. See [TrackStreamStateUpdatedEvent] for the
+  /// relevant event.
+  StreamState get streamState => _streamState;
+
+  @internal
+  Future<void> updateStreamState(StreamState streamState) async {
+    // return if no change
+    if (_streamState == streamState) return;
+    _streamState = streamState;
+    [participant.events, participant.room.events]
```
Comment (hiroshihorie) [inline lib/src/publication/remote_track_publication.dart:57]: It's firing at Room level here oniichan 😁 @boks1971

Comment (boks1971) [reply lib/src/publication/remote_track_publication.dart:57]: Got it, thank you thambi (I am going to start calling you thambi which is the opposite of oniichan in my mother tongue Tamil :-) )

**Comment by davidliu** [comment]
Yeah, I fire these at the room, participant, and track levels.

**Pull request merged**